server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -127,6 +127,9 @@
 
     }
 
+    public static final Setting<Integer> SETTING_CLUSTER_MAX_SHARDS_PER_NODE =
+        Setting.intSetting("cluster.shards.max_per_node", 1000, Property.Dynamic, Property.NodeScope);
```

Comment:
Can we set a minimum for this setting of 1 shard per node? (so that people don't set it to -171 and expect weird things)

Reply:
I wonder if a higher minimum is warranted -- e.g., to ensure if we are setting up a new cluster we can create a .kibana index and so on?

Reply:
The default here is a 1000 so we will be fine out of the box. The question here is the minimum and there is not a good value as there are many indices that might be created (.kibana, .security, Watcher, .monitoring, etc.). It is too hard to find the right minimum to ensure the basics of our stack function and to keep this value properly maintained. If someone really does want to set the minimum to one shard per node, I think we should permit that.

Reply:
Added a minimum of 1 - I agree with Jason, I think trying to figure out any other minimum would be very complicated.